src/api/app/models/project.rb

```diff
@@ -1164,20 +1164,20 @@ def open_requests
       )
     ).where(state: :review).distinct.order(priority: :asc, id: :desc).pluck(:number)
 
-    targets = BsRequest.with_involved_projects(id)
-                       .or(BsRequest.from_source_project(name))
+    targets = BsRequest.to_project(name)
```

Comment:
Here, instead of filtering by an specific id we switch to filter by a string with the LIKE  comparision operator. While the code looks cleaner this way, this could impact the performance of the query.

Reply:
So did you measure it?

Reply:
I did some measurements. First with the query made by `.with_involved_projects(id)`:

```
pry(main)> Benchmark.measure { BsRequest.where(bs_request_actions: { target_project_id: 173190 }) }.to_s
=> "  0.000206   0.000054   0.000260 (  0.000257)\n"
pry(main)> Benchmark.measure { BsRequest.where(bs_request_actions: { target_project_id: 173190 }) }.to_s
=> "  0.000275   0.000000   0.000275 (  0.000273)\n"
pry(main)> Benchmark.measure { BsRequest.where(bs_request_actions: { target_project_id: 173190 }) }.to_s
=> "  0.000362   0.000000   0.000362 (  0.000358)\n"
```
Then with the query made by `.to_project(name)`:

```
pry(main)> Benchmark.measure { BsRequest.where('bs_request_actions.source_project like ?', 'home:enavarro_suse') }.to_s
=> "  0.000124   0.000032   0.000156 (  0.000153)\n"
pry(main)> Benchmark.measure { BsRequest.where('bs_request_actions.source_project like ?', 'home:enavarro_suse') }.to_s
=> "  0.000000   0.000158   0.000158 (  0.000155)\n"
pry(main)> Benchmark.measure { BsRequest.where('bs_request_actions.source_project like ?', 'home:enavarro_suse') }.to_s
=> "  0.000000   0.000171   0.000171 (  0.000169)\n"
```

The columns are `user     system      total        real` times.

`total` times (third column) are better in the `.to_project(name)` option.

I guess the improvement comes from using Array conditions instead of Hash conditions.

Reply:
You see all the 0's in front of the numbers right? :)

Reply:
Sure. No problem with this one. This change is an improvement. :+1: